### ISSUE TITLE
Preprocessor statements in line continuations break syntax highlighting

### ISSUE THREAD

**Issue by rwirth** [krvajal/vscode-fortran-support#248]
My project contains variable declarations with attributes enabled or disabled by preprocessor statements:

```fortran
    REAL(wp), POINTER         &
#ifdef HAVE_FC_ATTRIBUTE_CONTIGUOUS
    , CONTIGUOUS              &
#endif
    &  ::                     &
```
This seems to confuse the syntax highlighter, switching off all highlighting until the next logical line:
![image](https://user-images.githubusercontent.com/3491963/125479277-ccca0edb-4983-4312-9b41-8ec8c50c821f.png)
Also, the code hints do not work for variables declared this way. But this might be a problem with the IntelliSense plugin or the language server.

**Comment by rwirth** [comment]
I didn't test it extensively, but adding 
```json
"source.fortran.free": {
   "patterns": [   
       {       
           "include": "#preprocessor"
       }
   ]
},
```
to the injections in the free-form syntax fixes the highlighting issue.

There is also an issue with the logical-AND operator in preprocessor lines getting misidentified as a Fortran line continuation operator. I believe this can be fixed by adding an identifier, e.g., `"name": "source.preprocessor"`, to repository->preprocessor, and adding that identifier to the list of exclusions in the injection for the continuation operator, viz., `"source.fortran.free - ( string | comment | source.preprocessor )"`.

**Comment by gnikit** [comment]
Good catch, I checked back with the original version of this documentation and the preprocessor directives should indeed be in the exclusion list.

Had completely forgotten about injections at the root level of the file.

Since I am at it, I will change the syntax highlight to match with C/C++
